test/src/tpm/tpm2_interface_test.c

```diff
@@ -692,29 +691,192 @@ void test_apply_policy(void)
+  init_pcr_selection("23", &pcrs_struct_3);
+  CU_ASSERT(pcrs_struct_3.count == 2);
+
+  //Start with empty policy digest list (no policy-OR)
+  TPML_DIGEST pOR_digests_struct = {.count = 0, };
+
   init_tpm2_connection(&sapi_ctx);
+  create_auth_session(sapi_ctx, &session, TPM2_SE_POLICY);
 
   //Valid test
-  create_auth_session(sapi_ctx, &session, TPM2_SE_POLICY);
-  TPML_PCR_SELECTION pcrs_struct = {.count = 0, };
-  CU_ASSERT(apply_policy(sapi_ctx, session.sessionHandle, pcrs_struct) == 0);
+  CU_ASSERT(apply_policy(sapi_ctx,
+                         session.sessionHandle,
+                         &(pcrs_struct_1.pcrs[0]),
+                         &pOR_digests_struct) == 0);
 
   //NULL context
-  CU_ASSERT(apply_policy(NULL, session.sessionHandle, pcrs_struct) != 0);
+  CU_ASSERT(apply_policy(NULL,
+                         session.sessionHandle,
+                         &(pcrs_struct_1.pcrs[0]),
+                         &pOR_digests_struct) != 0);
 
   //Invalid Handle
-  CU_ASSERT(apply_policy(sapi_ctx, 0, pcrs_struct) != 0);
+  CU_ASSERT(apply_policy(sapi_ctx,
+                         0,
+                         &(pcrs_struct_1.pcrs[0]),
+                         &pOR_digests_struct) != 0);
 
   //Multiple pcrs
-  int pcrs[2] = { };
-  pcrs[0] = 5;
-  pcrs[1] = 3;
-  init_pcr_selection(sapi_ctx, pcrs, 2, &pcrs_struct);
-  CU_ASSERT(apply_policy(sapi_ctx, session.sessionHandle, pcrs_struct) == 0);
+  CU_ASSERT(apply_policy(sapi_ctx,
+                         session.sessionHandle,
+                         &(pcrs_struct_2.pcrs[0]),
+                         &pOR_digests_struct) == 0);
+
+  //Policy-OR policy
+  //  - these tests require TPM2_ALG_SHA256 so we don't want to run them
+  //    if this changes
+  if (KMYTH_HASH_ALG == TPM2_ALG_SHA256)
+  { 
+    TPM2B_DIGEST policy1 = {.size = 0, };
+    TPM2B_DIGEST policy2 = {.size = 0, };
+    TPM2B_DIGEST policy3 = {.size = 0, };
+
+    // create policy digest for first policy branch
+    CU_ASSERT(create_policy_digest(sapi_ctx,
+                                   &(pcrs_struct_3.pcrs[0]),
+                                   &pOR_digests_struct,
+                                   &policy1) == 0);
+    CU_ASSERT(policy1.size == KMYTH_DIGEST_SIZE);
+    char policy1_str[(KMYTH_DIGEST_SIZE * 2) + 1];
+    convert_digest_to_string(&policy1, policy1_str);
+
+    if (system("tpm2_pcrextend 23:sha256=0000000000000000000000000000000000000000000000000000000000000001") != -1)
+    {
+      // create policy digest for second policy branch
+      CU_ASSERT(create_policy_digest(sapi_ctx,
+                                     &(pcrs_struct_3.pcrs[1]),
+                                     &pOR_digests_struct,
+                                     &policy2) == 0);
+      CU_ASSERT(policy2.size == KMYTH_DIGEST_SIZE);
+      char policy2_str[(KMYTH_DIGEST_SIZE * 2) + 1];
+      convert_digest_to_string(&policy2, policy2_str);
+      CU_ASSERT(strncmp(policy1_str,
```

Comment:
I believe that the issue here is a version dependency on TPM2 Tools. I have rewritten the tests for apply_policy() and apply_policy_or() to not require these system calls and, hence, eliminate the dependency on TPM2 Tools.

Also, in looking at these tests, I realized that they were written for the current policy-OR implementation and possibly only slightly modified because of the changes in this pull request. Therefore, I completely refactored the tests for apply_policy() and apply_policy_or() in an attempt to better exercise, test, and document this functionality.